### PR TITLE
Add splinecalib

### PR DESCRIPTION
**Pull request by philip-khor**
Required for ml-insights 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by philip-khor** [comment]
@conda-forge/help-python ready for review, thank you!

File: recipes/splinecalib/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (beenje) [inline recipes/splinecalib/LICENSE:1]: Could you please create an issue or PR upstream to request the LICENSE to be included in next release?

File: recipes/splinecalib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy >=1.16
```
Comment (beenje) [inline recipes/splinecalib/meta.yaml:22]: Remove the min version to let conda use the version specified in [pinning-feedstock](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml)

File: recipes/splinecalib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - scipy >=1.3
+  run:
+    - python
+    - numpy >=1.16
```
Comment (beenje) [inline recipes/splinecalib/meta.yaml:26]: Replace with:

`- {{ pin_compatible('numpy') }}`

File: recipes/splinecalib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+about:
+  home: https://github.com/numeristical/splinecalib
+  summary: 'Simple, fast, extensible JSON encoder/decoder for Python'
+  description: |
+    SplineCalib is a Python package for calibrating
+    ML models using smoothing splines. 
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  doc_url: https://splinecalib.readthedocs.io/
```
Comment (beenje) [inline recipes/splinecalib/meta.yaml:47]: This link doesn't work

File: recipes/splinecalib/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy >=1.16
+    - scipy >=1.3
```
Comment (beenje) [inline recipes/splinecalib/meta.yaml:23]: This isn't needed to build the package.

**Review by beenje — CHANGES_REQUESTED** [review]

File: recipes/splinecalib/LICENSE (hunk 1 above)
Comment (philip-khor) [reply recipes/splinecalib/LICENSE:1]: numeristical/splinecalib#1

File: recipes/splinecalib/meta.yaml (hunk 4 above)
Comment (philip-khor) [reply recipes/splinecalib/meta.yaml:47]: numeristical/splinecalib#2

File: recipes/splinecalib/meta.yaml (hunk 5 above)
Comment (philip-khor) [reply recipes/splinecalib/meta.yaml]: Removed scipy, please let me know if I've misunderstood.

**Pull request merged**